kratos/solving_strategies/convergencecriterias/mixed_generic_criteria.h

```diff
@@ -493,7 +493,8 @@ class MixedGenericCriteria : public ConvergenceCriteria< TSparseSpace, TDenseSpa
                     dof_dx = TSparseSpace::GetValue(rDx, dof_id);
 
                     const auto &r_current_variable = it_dof->GetVariable();
-                    int var_local_key = mLocalKeyMap[r_current_variable.IsComponent() ? r_current_variable.GetSourceVariable().Key() : r_current_variable.Key()];
+                    const KeyType key = r_current_variable.IsComponent() ? r_current_variable.GetSourceVariable().Key() : r_current_variable.Key();
+                    const int var_local_key = mLocalKeyMap.at(key);
```

Comment:
please use `find` with a corresponding proper error. My issue with `at` is that it only throws the cryptic standard lib error

Reply:
cool, that makes sense.

Reply:
changed, feel free to suggest another error message, Im very bad at those :)